source/upgrade.txt

```diff
@@ -0,0 +1,104 @@
+.. _java-upgrade-driver:
+
+=======================
+Upgrade Driver Versions
+=======================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 1
+   :class: singlecol
+
+
+Overview
+--------
+
+In this section, you can identify the changes you need to make to your
+application to upgrade your driver to a new version.
+
+Before you upgrade, perform the following actions:
+
+- Ensure the new version is compatible of with the MongoDB server versions
+  your application connects to and the Java Runtime Environment (JRE) your
+  application runs on. See the :ref:`Java Compatibility <java-compatibility-tables>`
+  page for this information.
+- Address any breaking changes between the current version of the driver
+  your application is using and your planned upgrade version in the
+  section on :ref:`Breaking Changes <java-breaking-changes>` below.
+
+.. tip::
+
+   To minimize the amount of changes your application may require when
+   upgrading driver versions in the future, use the
+   :ref:`{+stable-api+} <stable-api-java>`.
+
+.. _java-breaking-changes:
+
+Breaking Changes
+----------------
+
+A breaking change is a modification in a convention or behavior in
+a specific version of the driver that may prevent your application from
+working properly if not addressed before upgrading.
+
+The breaking changes in this section are categorized by the driver version that
+introduced them. When upgrading driver versions, address all the breaking
+changes between the current and upgrade versions. For example, if you
+are upgrading the driver from v4.0 to v4.5, address all breaking changes from
+the version after v4.0 including any listed under v4.5.
+
+.. _java-breaking-changes-v4.2:
+
+Version 4.2 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- The ``ObjectId`` class and its ``serialVersionUID`` field were updated
+  to use a new format that minimizes serialization compatibility issues
+  across different versions of the driver.
+
+  Any objects that include an ``ObjectId`` field that were serialized with an
```

Comment:
I'm still concerned about this. Many users will think we're talking about serializing into and out of MongoDB, not into and out of a Java ObjectInputStream/ObjectOutputStream.

Also, technically it's not the driver that actually throws an exception, it's the JDK

Reply:
> I'm still concerned about this. Many users will think we're talking about serializing into and out of MongoDB, not into and out of a Java ObjectInputStream/ObjectOutputStream.

Ok, I'll remove it unless you have something specific you want to include.